packages/pg-v5/lib/psql.js

```diff
@@ -49,9 +49,22 @@ function execPsqlWithFile (file, dbEnv) {
 function psqlInteractive (dbEnv, prompt) {
   const {spawn} = require('child_process')
   return new Promise((resolve, reject) => {
-    let psql = spawn('psql',
-      ['--set', `PROMPT1=${prompt}`, '--set', `PROMPT2=${prompt}`],
-      {env: dbEnv, stdio: 'inherit'})
+    let psql_args = ['--set', `PROMPT1=${prompt}`, '--set', `PROMPT2=${prompt}`, ]
+    let path = process.env.HEROKU_PG_HISTORY
+    if(path) {
+      const fs = require('fs')
+      if(fs.existsSync(path) && fs.statSync(path).isDirectory()) {
+        psql_args = psql_args.concat(['--set', `HISTFILE=${path}/${prompt.split(':')[0]}`])
+      }
+      else if (fs.existsSync(require('path').dirname(path))) {
```

Comment:
@mage2k I recommend moving the `require('path')` up into it's own line below `const fs = require('fs')`.  I find it busy.  I also recommend renaming `path` to something more specific e.g. `pgHistoryPath`.